src/test/java/com/atlassian/db/replica/api/ThrottledCacheTest.java

```diff
@@ -113,11 +114,58 @@ public void shouldNotRobbedThreadReleaseLock() throws InterruptedException {
         assertThat(throwable).doesNotThrowAnyException();
     }
 
+    @Test
+    public void shouldntRobbedThreadUpdateValue() throws InterruptedException {
+        ThrottledCache<Long> cache = new ThrottledCache<>(clock, ofMillis(1500));
+
+        final WaitingWork firstThreadWork = asyncPutWithSlowSupplier(cache, 1);
+        clock.tick();
+        clock.tick();
+        cache.get(() -> 2L);
+        firstThreadWork.finish();
+        assertThat(cache.get()).isEqualTo(Optional.of(2L));
+    }
+
+    @Test
+    public void shouldFailingSupplierReleaseTheLock() throws InterruptedException {
+        ThrottledCache<Long> cache = new ThrottledCache<>(clock, ofMillis(1500));
+
+        cache.get(() -> 1L);
+        try {
+            cache.get(() -> {
+                throw new RuntimeException();
+            });
+        } catch (Exception e) {
+            //ignore
```

Comment:
for what `//ignore` refers to? exception or test scenario?

Reply:
I refactored it.